src/elfpy/simulators.py

```diff
@@ -304,38 +296,12 @@ def run_simulation(self, override_dict=None):
             # Could define a 'trade amount & direction' time series that's a function of the vault apy
             # Could support using actual historical trades or a fit to historical trades)
             day_trading_volume = 0
-            days_trades = []
             while day_trading_volume < self.target_daily_volume:
-                if self.pool_apy_target_range is not None:
-                    pool_apy = self.market.apy(self.get_days_remaining())
-                    (
-                        token_index,
-                        self.apy_distance_in_target_range,
-                        self.apy_distance_from_mid_when_in_range,
-                        self.actual_convergence_strength,
-                        self.expected_proportion,
-                        self.streak_luck,
-                        btest,
-                    ) = User.stochastic_direction(
-                        pool_apy=pool_apy,
-                        pool_apy_target_range=self.pool_apy_target_range,
-                        days_trades=days_trades,
-                        pool_apy_target_range_convergence_speed=self.pool_apy_target_range_convergence_speed,
-                        rng=self.rng,
-                        run_trade_number=self.run_trade_number,
-                        verbose=self.verbose,
-                    )
-                else:
-                    token_index = User.random_direction(self.rng)
-                days_trades.append(token_index)
-                self.token_in = self.tokens[token_index]
-                self.token_out = self.tokens[1 - token_index]
-                self.trade_amount_usd = self.get_trade_amount_usd(
-                    self.token_in,
-                    self.trade_direction,
-                    self.target_daily_volume,
-                    self.base_asset_price,
+                (self.token_in, self.token_out, self.trade_amount_usd) = self.user.get_trade(
+                    self.market, self.tokens, self.trade_direction, self.target_daily_volume, self.base_asset_price
                 )
+                if self.user_type.lower() == "weightedrandom":  # sync user with market
+                    self.user.set_market_apy(self.market.apy(self.get_days_remaining()))
```

Comment:
why are we calling this from the simulators class? the user could update this themselves by accessing the market, no? is it more efficient this way, since we only need to update once after we know the apy changed, as opposed to calling the function every time the apy is referenced?

Reply:
You set it up so that the weighting in WeightedRandom is a function of the Market APY. This was to avoid having to pass the entire market object to the User class, which was previously done only for that one variable.

Reply:
right, i thought we were still doing that for a moment. now we pass just the apy, which is much more lightweight.